e2e/bitrise.yml

```diff
@@ -214,19 +215,6 @@ workflows:
     - _check_outputs
     - _teardown
 
-  test_checkout_pull_request_standard_branch:
```

Comment:
I don't think this test should have ever existed. It doesn't make sense to pass a branch name as the merge ref.
I also checked our webhook processor code, it seems to be only used with GitHub events, and it uses GitHub's merge ref format: https://github.com/search?q=repo:bitrise-io/bitrise-webhooks%20PullRequestMergeBranch&type=code

Reply:
In theory, would we need this functionality for git providers that don't provide a merge ref?

Reply:
When the merge ref is not available, the PR branch is still available (for example, `pull/7/head`) and the step is going to fetch and checkout that.

Reply:
Sounds good 👍

Reply:
Minor correction: I learned that sometimes even the PR head ref (`pull/7/head`) is unavailable (out of the major Git providers we support, Bitbucket is the only one not exposing this), but the PR branch name is always available and the step uses that as the fallback.

In merge requests, there is no real difference between the MR head and the MR branch, but for true pull requests, the PR head points to the destination repo while the PR branch points to the fork repo.